### PR TITLE
Add globalDCE and possibly globalsAA to the pipeline

### PR DESCRIPTION
**Pull request by gbaraldi**
This removes the unnecessary globals from the module making the IR cleaner and smaller

### CODE REVIEW

**Review by vtjnash — APPROVED** [review]

File: src/pipeline.cpp [hunk 1]
```diff
@@ -383,6 +387,7 @@ static void buildEarlyOptimizerPipeline(ModulePassManager &MPM, PassBuilder *PB,
             FPM.addPass(SROAPass());
             // SROA can duplicate PHI nodes which can block LowerSIMD
             FPM.addPass(InstCombinePass());
+            MPM.addPass(GlobalDCEPass());
```
Comment (vtjnash) [inline src/pipeline.cpp:390]: this should probably be part of O1 also, and maybe O0 too

**Review by vtjnash — COMMENTED** [review]
IIUC, this is supposed to be kept the same as `addOptimizationPasses`, until we delete that other code?

**Comment by vchuravy** [comment]
Yeah, we only have the old code around for GPUCompiler

**Comment by gbaraldi** [comment]
Hopefully we can move LLVM.jl and GPUCompiler to the newPM so that we can remove this code.

**Comment by gbaraldi** [comment]
Bump!

**Pull request merged**